jhipster-framework/src/main/java/io/github/jhipster/security/management/ApiKeyAuthenticationFilter.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2016-2019 the original author or authors from the JHipster project.
+ *
+ * This file is part of the JHipster project, see https://www.jhipster.tech/
+ * for more information.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.github.jhipster.security.management;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.util.StringUtils;
+import org.springframework.web.filter.GenericFilterBean;
+
+/**
+ * Filters incoming requests and install the "manager" Spring Security principal if the Authorization header
+ * contains the correct API Key.
+ */
+public class ApiKeyAuthenticationFilter extends GenericFilterBean {
+    static final String AUTHORIZATION_HEADER = "Authorization";
+    static final String MANAGER = "manager";
```

Comment:
This user doesn't really exist (eg. in DB). I find it weird...

Reply:
I agree that it's strange but then this user need a name.